theano/tensor/nlinalg.py

```diff
@@ -193,75 +193,6 @@ def infer_shape(self, node, shapes):
         return [(x_s[0], x_s[0])]
 
 alloc_diag = AllocDiag()
-
-
-class ExtractDiag(Op):
-    """Return the diagonal of a matrix.
-
-    Notes
-    -----
-    Works on the GPU.
-
-    """
-
-    __props__ = ("view",)
-
-    def __init__(self, view=False):
-        self.view = view
-        if self.view:
-            self.view_map = {0: [0]}
-
-    def make_node(self, _x):
-        warnings.warn("DeprecationWarning: theano.tensor.nlinalg.ExtractDiag"
-                      "is deprecated, please use theano.tensor.ExtractDiag"
-                      "instead.",
-                      category=DeprecationWarning)
-        if not isinstance(_x, theano.Variable):
-            x = as_tensor_variable(_x)
-        else:
-            x = _x
-
-        if x.type.ndim != 2:
-            raise TypeError('ExtractDiag only works on matrices', _x)
-        y = x.type.clone(broadcastable=(False,))()
-        return Apply(self, [x], [y])
-
-    def perform(self, node, ins, outs):
-        """ For some reason numpy.diag(x) is really slow, so we
-        implemented our own. """
```

Comment:
@nouiz , the perform function in here is different from the one in tensor/basic.  It says numpy's diagonal is slow and has its own implementation. Do we want to remove this code?

Reply:
This is for an old version of NumPy. I think we can get rid of this code now.